srf_generation/input_file_generation/realisation_to_srf.py

```diff
@@ -0,0 +1,292 @@
+    lon=None,
+    lat=None,
+    shypo=None,
+    dhypo=None,
+    mwsr=None,
+    tect_type=None,
+    dip_dir=None,
+    file_name=None,
+):
+    """
+    Stores SRF metadata as hdf5.
+    srf_file: SRF path used as basename for info file and additional metadata
+    """
+    planes = srf.read_header(srf_file, idx=True)
+    hlon, hlat, hdepth = srf.get_hypo(srf_file, depth=True)
+
+    dbottom = []
+    corners = np.zeros((len(planes), 4, 2))
+    for i, p in enumerate(planes):
+        # currently only support single dip dir value
+        if dip_dir is not None:
+            dip_deg = dip_dir
+        else:
+            dip_deg = p["strike"] + 90
+        # projected fault width (along dip direction)
+        pwid = p["width"] * np.cos(np.radians(p["dip"]))
+        corners[i, 0] = geo.ll_shift(
+            p["centre"][1], p["centre"][0], p["length"] / 2.0, p["strike"] + 180
+        )[::-1]
+        corners[i, 1] = geo.ll_shift(
+            p["centre"][1], p["centre"][0], p["length"] / 2.0, p["strike"]
+        )[::-1]
+        corners[i, 2] = geo.ll_shift(corners[i, 1, 1], corners[i, 1, 0], pwid, dip_deg)[
+            ::-1
+        ]
+        corners[i, 3] = geo.ll_shift(corners[i, 0, 1], corners[i, 0, 0], pwid, dip_deg)[
+            ::-1
+        ]
+        dbottom.append(p["dtop"] + p["width"] * np.sin(np.radians(p["dip"])))
+
+    if file_name is None:
+        file_name = srf_file.replace(".srf", ".info")
+
+    with h5open(file_name, "w") as h:
+        a = h.attrs
+        # only taken from given parameters
+        a["type"] = srf_type
+        a["dt"] = dt
+        a["rake"] = rake
+        a["mag"] = mag
+        # srf header data
+        for k in planes[0].keys():
+            a[k] = [p[k] for p in planes]
+        # point source has 1 vs/rho, others are taken from vm
+        if srf_type == 1:
+            a["vs"] = vs
+            a["rho"] = rho
+            a["hlon"] = lon
+            a["hlat"] = lat
+            a["hdepth"] = centroid_depth
+        else:
+            a["vm"] = np.string_(path.basename(vm))
+            a["hlon"] = hlon
+            a["hlat"] = hlat
+            a["hdepth"] = hdepth
+            a["shyp0"] = shypo
+            a["dhyp0"] = dhypo
+        if mwsr is not None:
+            a["mwsr"] = np.string_(mwsr)
+        if tect_type is not None:
+            a["tect_type"] = tect_type
+        # derived parameters
+        a["corners"] = corners
+        a["dbottom"] = dbottom
+
+
+def create_ps_srf(cs_root: str, parameter_dictionary: Dict[str, Any]):
+    name = parameter_dictionary.get("name")
+
+    latitude = parameter_dictionary.pop("latitude")
+    longitude = parameter_dictionary.pop("longitude")
+    depth = parameter_dictionary.pop("depth")
+    magnitude = parameter_dictionary.pop("magnitude")
+    strike = parameter_dictionary.pop("strike")
+    rake = parameter_dictionary.pop("rake")
+    dip = parameter_dictionary.pop("dip")
+    moment = parameter_dictionary.pop("moment", 0)
```

Comment:
I presume pop gives you a default value. That's a cool usage.

Reply:
Nope, pop will raise an Exception if it doesn't have a second parameter to supply as a default. Optional parameters have the current default provided.

Reply:
That's what I meant lol. Very cool.